lib/pbench/server/api/resources/query_apis/datasets/table_of_content.py

```diff
@@ -0,0 +1,218 @@
+from http import HTTPStatus
+from logging import Logger
+
+from flask_restful import abort
+
+from pbench.server import PbenchServerConfig
+from pbench.server.api.resources import (
+    JSON,
+    Schema,
+    Parameter,
+    ParamType,
+)
+from pbench.server.api.resources.query_apis import CONTEXT
+from pbench.server.api.resources.query_apis.datasets import RunIdBase
+
+
+class TableOfContent(RunIdBase):
+    """
+    Table of Content API returns the list of sub-directories and files
+    present under a directory.
+    """
+
+    MAX_SIZE = 10000
+    parent = None
```

Comment:
Now that you're caching the parent value in the context parameter, there should be no need for this class attribute.  (In fact, I don't see where it ever gets assigned a value other than here, which makes me wonder how the code is actually working...did you miss a commit?)

Reply:
yeah